lib/install/management/delete.go

```diff
@@ -104,11 +104,14 @@ func (d *Dispatcher) DeleteVCHInstances(vmm *vm.VirtualMachine, conf *metadata.V
 		err = errors.Errorf("Failed to get VCH resource pool (%s): %s", rpRef, err)
 		return err
 	}
-	_, ok := ref.(*object.ResourcePool)
-	if !ok {
-		log.Errorf("Failed to find resource pool %s, %s", rpRef, err)
-		return err
+	switch ref.(type) {
+	case *object.VirtualApp:
+	case *object.ResourcePool:
+		break
```

Comment:
No need for the 'break' here, since Go switch does not fall through by default.  You could even just do:
```
switch ref.(type) {
    case *object.VirtualApp, *object.ResourcePool:
        // ok
    default:
        log.Errorf(...)
}
```